tracer/src/Datadog.Trace.SourceGenerators/Helpers/IncrementalGeneratorBase.cs

```diff
@@ -0,0 +1,90 @@
+// <copyright file="IncrementalGeneratorBase.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using Datadog.Trace.SourceGenerators;
+using Datadog.Trace.SourceGenerators.Helpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Helpers
+{
+    /// <summary>
+    /// Base class
+    /// </summary>
+    public abstract class IncrementalGeneratorBase : IIncrementalGenerator, IAdditionalFileCodeGenerator
+    {
+        /// <summary>
+        /// Gets or sets set additional file writer function
+        /// </summary>
+        public Action<string, string> WriteAdditionalFile { get; set; } = DefaultWriteAdditionalFile;
+
+        /// <summary> Init </summary>
+        /// <param name="context"> Ctx </param>
+        public abstract void Initialize(IncrementalGeneratorInitializationContext context);
+
+        internal static Result<string> GetTfm(GeneratorAttributeSyntaxContext context, CancellationToken ct)
```

Comment:
You're not actually returning any errors in the `Result`, so there's no value to using it, just return `string`. I also don't think you want the `ERROR` cases, right? so i would return `null` for those instead, and filter them out in `RegisterTfm()`

Reply:
You are right

Reply:
Fixed in af7572f49550f1c3dbaad95fb44834e6b021784f